packages/esm-patient-common-lib/src/siderail-action-button/siderail-action-button.component.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Button } from '@carbon/react';
+import { useLayoutType } from '@openmrs/esm-framework';
+import { useWorkspaces, useWorkspaceWindowSize } from '../workspaces';
+import styles from './siderail-action-button.scss';
+
+interface SiderailActionButtonProps {
+  getIcon: (props: Object) => JSX.Element;
+  label: string;
+  iconDescription: string;
+  handler: () => void;
+  workspaceMatcher: RegExp | string | ((name: string) => boolean);
```

Comment:
Why not just use the workspace "type"?

Reply:
The matching is actually being done on the name below.

Actually, by default the `form` type is common for all workspaces, hence the action button will not work properly when it comes to clinical form. Rather than using the workspace type, which is by default `form` for all workspaces, matching the name will be a better usecase.

What do you think about this?

Reply:
`type` was literally invented for this purpose. I think you should fix the clinical form's type; it was a mistake for whoever created it to leave the type as "form," since it has its own button. The default is "form" because it is the most common—e.g. the vitals form, allergies form, etc. all have that type.